README.md

```diff
@@ -1,72 +1,183 @@
+:exclamation: **NOTE:** This project is in preview and breaking changes are 
+introduced frequently. Therefore, vendoring dependencies is even more
+important than usual. We use [dep](https://github.com/golang/dep).
 
-# Installation
-- If you don't already have it, install [the Go Programming Language](https://golang.org/dl/).
-- Go get the SDK:
+To be notified about updates and changes, subscribe to the [Azure update feed][].
 
-```
+**Install:**
+
+```sh
 $ go get -u github.com/Azure/azure-sdk-for-go/...
 ```
 
-> **IMPORTANT:** We highly suggest vendoring Azure SDK for Go as a dependency. For vendoring dependencies, Azure SDK
-for Go uses [dep](https://github.com/golang/dep).
+If you need to install Go, follow [the official instructions][].
+
+[the official instructions]: https://golang.org/dl/
+
+**Use:**
+
+For complete examples see [Azure-Samples/azure-sdk-for-go-samples][samples_repo].
+
+1. Import a package from the [services][services_dir] directory. See [SERVICES.md][] for
+   a list.
+1. Create and authenticate a client with a `New*Client` func, e.g.
+   `c := compute.NewVirtualMachinesClient(...)`.
+1. Invoke API methods using the client, e.g. `c.CreateOrUpdate(...)`.
+1. Handle responses.
+
+For example, to create a new virtual network (substitute your own values for
+strings in angle brackets):
+
+```go
+import (
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+vnetClient := network.NewVirtualNetworksClient("<subscriptionID>")
+vnetClient.Authorizer = autorest.NewBearerAuthorizer("<OAuthTokenProvider>")
+
+vnetClient.CreateOrUpdate(
+  "<resourceGroupName>",
+  "<vnetName>",
+  network.VirtualNetwork{
+    Location: to.StringPtr("<azureRegion>"),
+    VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
+      AddressSpace: &network.AddressSpace{
+        AddressPrefixes: &[]string{"10.0.0.0/8"},
+      },
+      Subnets: &[]network.Subnet{
+        {
+          Name: to.StringPtr("<subnet1Name>"),
+          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+            AddressPrefix: to.StringPtr("10.0.0.0/16"),
+          },
+        },
+        {
+          Name: to.StringPtr("<subnet2Name>"),
+          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+            AddressPrefix: to.StringPtr("10.1.0.0/16"),
+          },
+        },
+      },
+    },
+  },
+  nil)
+```
+
+## Authentication
+
+Most operations require an OAuth token for authentication and authorization.
+Following is how to get one using the
+[adal](https://github.com/Azure/go-autorest/tree/master/autorest/adal)
+package.
+
+If you need a service principal (aka Client, App), run `az ad sp create-for-rbac -n "<app_name>"` 
+in [azure-cli](https://github.com/Azure/azure-cli), see
+[docs](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest
```

Comment:
Incorrectly formatted MD url, also the text is difficult to parse. Would suggest:

````
Most operations require authentication. The following snippet shows how to get one using the [adal](...) package,
as long as you have an available service principal. If you need to create a service principal, you can do so with the
[Azure CLI 2.0](https://docs.microsoft.com/en-us/cli/azure/overview) with the command:

```
... command here ...
```

After creating the service principal, write down the ID, secret, and tenant ID to use in your Go code.
````

Even this may not provide enough context to be useful. We should include a link to the full service principal docs at some point because they cover things like the differentiation between roles and when to use what.

Reply:
Cleaned up language, PTAL. Thanks!